miracl/flow/miracl_workflow_ace_interface.py

```diff
@@ -100,46 +115,49 @@ def convert(self, args):
 
 
 class ACERegistration(Registration):
-    def register(self, args, **kwargs):
-        print("  registering...")
-        if "dependent_folder" in kwargs:
-            ace_flow_conv_output_folder = kwargs["dependent_folder"]
-        else:
-            raise FileNotFoundError("Output folder path variable not found!")
+    """ACE Registration module used for registering the converted nifti files to the Allen atlas.
 
-        if "converted_nii_file" in kwargs:
-            converted_nii_file = kwargs["converted_nii_file"]
-        else:
-            raise FileNotFoundError("Converted nifti file not found!")
+    :param Registration: base abstract class for registration
+    :type Registration: ABC
+    """
+
+    def register(self, args: argparse.Namespace, reg_cmd: str):
+        """Main method for registration module. Calls `miracl reg clar_allen` module.
+
+        :param args: command line arguments needed for MIRACL reg module.
+        :type args: argparse.Namespace
+        :param reg_cmd: registration command with args.
+        :type reg_cmd: str
+        """
 
-        reg_cmd = f"{MIRACL_HOME}/reg/miracl_reg_clar-allen.sh \
-        -i {converted_nii_file} \
-        -r {args.sa_output_folder} \
-        -o {args.rca_orient_code} \
-        -m {args.rca_hemi} \
-        -v {args.rca_voxel_size} \
-        -l {args.rca_allen_label} \
-        -a {args.rca_allen_atlas} \
-        -s {args.rca_side} \
-        -f {args.rca_no_mosaic_fig} \
-        -b {args.rca_olfactory_bulb} \
-        -p {args.rca_skip_cor} \
-        -w {args.rca_warp}"
         subprocess.Popen(reg_cmd, shell=True).wait()
         logger.debug("Calling registration fn here")
         logger.debug(f"Example args: {args.rca_allen_atlas}")
-        logger.debug(f"dependent_folder: {ace_flow_conv_output_folder}")
-        logger.debug(f"nifti file: {converted_nii_file}")
+        # logger.debug(f"dependent_folder: {ace_flow_conv_output_folder}")
+        # logger.debug(f"nifti file: {converted_nii_file}")
 
 
 class ACEVoxelization(Voxelization):
-    def voxelize(self, args, stacked_tif):
+    """ACE Voxelization module used for voxelizing the segmented tif files.
+
+    :param Voxelization: base abstract class for voxelization
+    :type Voxelization: ABC
+    """
+
+    def voxelize(self, args: argparse.Namespace, stacked_tif: pathlib.Path):
+        """Main method for voxelization module. Calls `miracl seg voxelize` module.
+
+        :param args: command line arguments needed for MIRACL seg module.
+        :type args: argparse.Namespace
+        :param stacked_tif: path to the stacked tif file ('stacked_seg_tif.tif')
+        :type stacked_tif: pathlib.Path
+        """
         x_vox, y_vox, z_vox = args.sa_resolution
         print("  voxelizing stacked tif...")
         vox_cmd = f"miracl seg voxelize \
         --seg {stacked_tif} \
         --res {args.rca_voxel_size} \
-        --down {args.ctn_down} \
```

Comment:
This is what I spoke about with @AAttarpour. We can have different `conv` downsample and `vox` downsample args.